pkg/manager/member/tikv_member_manager.go

```diff
@@ -587,7 +587,13 @@ func (tkmm *tikvMemberManager) syncTidbClusterStatus(tc *v1alpha1.TidbCluster, s
 		return err
 	}
 
+	addressPrefix := fmt.Sprintf("%s-tikv-", tc.Name)
+	addressSuffix := fmt.Sprintf("%s-tikv-peer.%s.svc:20160", tc.Name, tc.Namespace)
 	for _, store := range storesInfo.Stores {
+		// make sure that the store belongs to the tikv
+		if store.Store != nil && !(strings.HasPrefix(store.Store.Address, addressPrefix) && strings.HasSuffix(store.Store.Address, addressSuffix)) {
+			continue
```

Comment:
- we should not hard code `20160` port here. although we didn't support it yet, the port number is configurable in theory, `net.SplitHostPort` can help to extract the host from the address
- I suggest using the regex to do the checking, e.g. check if the host matches the pattern `<cluster>-tikv-\d+-tikv-peer.<namespace>.svc`,

Reply:
by the way, do tiflash and tikv in the same cluster have other properties we can use to distinguish them?

Reply:
> by the way, do tiflash and tikv in the same cluster have other properties we can use to distinguish them?

I think the address might be the best property we can use to distinguish them. Here is the store info example for the tidbcluster.
```json
{
      "store": {
        "id": 6,
        "address": "tidb-tikv-0.tidb-tikv-peer.tidb-admin.svc:20160",
        "labels": [
          {
            "key": "host",
            "value": "docker-desktop"
          }
        ],
        "version": "3.0.8",
        "state_name": "Up"
      },
      "status": {
        "capacity": "465.6GiB",
        "available": "187.5GiB",
        "leader_count": 7,
        "leader_weight": 1,
        "leader_score": 7,
        "leader_size": 7,
        "region_count": 21,
        "region_weight": 1,
        "region_score": 21,
        "region_size": 21,
        "start_ts": "2020-02-04T05:43:43Z",
        "last_heartbeat_ts": "2020-02-04T05:51:13.2362298Z",
        "uptime": "7m30.2362298s"
      }
    }

```

Reply:
> * we should not hard code `20160` port here. although we didn't support it yet, the port number is configurable in theory, `net.SplitHostPort` can help to extract the host from the address
> * I suggest using the regex to do the checking, e.g. check if the host matches the pattern `<cluster>-tikv-\d+\.<cluster>-tikv-peer\.<namespace>\.svc`,

updated.

Reply:
For tiflash, we have "engine=tiflash" in store.labels.
But not guarantee for other situation such as "external tikv"

```
    {
      "store": {
        "id": 44,
        "address": "172.16.5.85:9512",
        "labels": [
          {
            "key": "tiflash_http_port",
            "value": "4512"
          },
          {
            "key": "engine",
            "value": "tiflash"
          }
        ],
        "version": "3.1.0-beta",
        "peer_address": "172.16.5.85:9012",
        "state_name": "Up"
      },
      "status": {
        "capacity": "1.063TiB",
        "available": "1.063TiB",
        "used_size": "42MiB",
        "leader_weight": 1,
        "region_count": 1,
        "region_weight": 1,
        "region_score": 1,
        "region_size": 1,
        "start_ts": "2020-02-27T13:15:29+08:00",
        "last_heartbeat_ts": "2020-02-27T13:25:59.825627093+08:00",
        "uptime": "10m30.825627093s"
      }
    }
```

Reply:
Cool, this would be pretty helpful.